backend/siarnaq/api/episodes/models.py

```diff
@@ -270,12 +283,84 @@ class Tournament(models.Model):
+        )
+
+    def initialize(self):
         """
-        Seed the tournament with eligible teamsn in order of decreasing rating, and
-        populate the Challonge brackets.
+        Seed the tournament with eligible teams in order of decreasing rating,
+        populate the Challonge brackets, and create TournamentRounds.
         """
-        raise NotImplementedError
+
+        tour_name_public = f"{self.episode.name_long} {self.name_long}"
+        tour_name_private = tour_name_public + " (private)"
+
+        # For security by obfuscation,
+        # and to allow easy regeneration of bracket
+        key = random.randint(1000, 9999)
+        # Challonge does not allow hyphens in its IDs
+        # so substitute them just in case
+        tour_id_public = (f"{self.episode.name_short}_{self.name_short}_{key}").replace(
+            "-", "_"
+        )
+        tour_id_private = f"{tour_id_public}_private"
+
+        # TODO support double
+        is_single_elim = True
+        participants = self.get_potential_participants()
+        # Parse into a format Challonge enjoys
+        # 1-idx seed
+        # Store team id in misc, for convenience (re-looking up is annoying)
+        # Store tour submission in misc, for consistency and convenience
+        # Note that tour submission should never change during a tournament anyways
+        # due to submission freeze. Bad things might happen if it does tho
+        participants = [
+            {"name": p.name, "seed": idx + 1, "misc": f"{p.id},{p.active_submission}"}
+            for (idx, p) in enumerate(participants)
+        ]
+        print(participants)
+
+        # First bracket made should be private,
+        # to hide results and enable fixing accidents
+        challonge.create_tour(tour_id_private, tour_name_private, True, is_single_elim)
+        challonge.bulk_add_participants(tour_id_private, participants)
+        challonge.start_tour(tour_id_private)
+
+        tour = json.loads(challonge.get_tour(tour_id_private))
+        # Derive round IDs
+        # Takes some wrangling with API response format
+        # TODO move this block to challonge.py
+        rounds = set()
+        for item in tour["included"]:
+            if item["type"] == "match":
+                round_idx = item["attributes"]["round"]
+                if round_idx not in rounds:
+                    rounds.add(round_idx)
+
+        # TODO check and tweak iter order in double elim
+        round_objects = [
+            TournamentRound(
+                tournament=self,
+                challonge_id=round_idx,
+                name=f"{tour_name_private} Round {round_idx}",
+            )
+            for round_idx in rounds
+        ]
+        print(round_objects)
+        TournamentRound.objects.bulk_create(round_objects)
+
+        self.challonge_id_private = tour_id_private
+        self.challonge_id_public = tour_id_public
+        self.in_progress = True
+        self.save()
```

Comment:
Minor optimization: use `update_fields` kwarg on save.

Reply:
track for later